src/main/java/org/folio/rest/client/InventoryClient.java

```diff
@@ -0,0 +1,109 @@
+package org.folio.rest.client;
+
+import static io.vertx.core.Future.failedFuture;
+import static io.vertx.core.Future.succeededFuture;
+import static javax.ws.rs.core.HttpHeaders.ACCEPT;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.folio.rest.jaxrs.model.HoldingsRecords;
+import org.folio.rest.jaxrs.model.Items;
+
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.ext.web.client.HttpRequest;
+import io.vertx.ext.web.client.HttpResponse;
+import io.vertx.ext.web.client.WebClient;
+
+public class InventoryClient extends OkapiClient {
+
+  private static final String ITEMS_LIMIT = "1000";
+  private static final String HOLDINGS_LIMIT = "1000";
+
+  public InventoryClient(WebClient webClient, Map<String, String> okapiHeaders) {
+    super(webClient, okapiHeaders);
+  }
+
+  public Future<Items> getItemsById(List<String> itemIds) {
+    if (itemIds.size() == 0) {
+      return Future.succeededFuture(new Items()
+        .withItems(new ArrayList<>())
+        .withTotalRecords(0));
+    }
+
+    Promise<HttpResponse<Buffer>> promise = Promise.promise();
+
+    HttpRequest<Buffer> request = okapiGetAbs("/inventory/items");
+    if (!itemIds.isEmpty()) {
+      request.addQueryParam("query", String.format("(id==(%s))",
+        itemIds.stream()
+          .map(id -> String.format("\"%s\"", id))
+          .collect(Collectors.joining(" or "))))
+      .addQueryParam("limit", ITEMS_LIMIT);
+    }
+    request.send(promise);
```

Comment:
We should create a tech debt issue about splitting this call into multiple calls in case the length of the URL exceeds server's limit.